### PR TITLE
allow users to define initialization weights as LookupTableData or an…

### PR DESCRIPTION
**Pull request by rmudambi**
… artifact key

### Description

- *Category*: feature
- *JIRA issue*: https://jira.ihme.washington.edu/browse/MIC-5580

### Changes and notes
- Allow users to define initialization weights as LookupTableData or an artifact key

### Testing
Tests pass

### CODE REVIEW

**Review by albrja — APPROVED** [review]

File: src/vivarium/framework/state_machine.py [hunk 1]
```diff
@@ -236,6 +239,20 @@ def __init__(
     # Public methods #
     ##################
 
+    def is_initial_state(self) -> bool:
+        """Determines if simulants could be initialized into this state.
+
+        Note: this will incorrectly return True if initialization_weights is a
```
Comment (stevebachmeier) [inline src/vivarium/framework/state_machine.py:245]: This kinda seems like an important bug we shouldn't introduce. Is this case handled downstream at all?

**Review by stevebachmeier — COMMENTED** [review]
I'm concerned about the case where an artifact or function returns all 0s

File: src/vivarium/framework/state_machine.py (hunk 1 above)
Comment (rmudambi) [reply src/vivarium/framework/state_machine.py:245]: The only implication of this happening is that we will throw a validation error if you specify an `initial_state` for your model if you specify a callable or artifact key for any state in that model even if that callable will always return 0, or if the key in the artifact only contains 0s. This doesn't seem like a big deal to me.

**Review by hussain-jafari — APPROVED** [review]

File: src/vivarium/framework/state_machine.py (hunk 1 above)
Comment (stevebachmeier) [reply src/vivarium/framework/state_machine.py:245]: I'd recommend at least pointing out in the docstring that it's not a big deal

File: src/vivarium/framework/state_machine.py [hunk 2]
```diff
@@ -236,6 +239,20 @@ def __init__(
...
+        callable or an artifact key that will always return 0.
+        """
+        if self.initialization_weights is None:
```
Comment (stevebachmeier) [inline src/vivarium/framework/state_machine.py:248]: I don't think you need these two lines

**Review by stevebachmeier — APPROVED** [review]

**Review by stevebachmeier — APPROVED** [review]

**Pull request merged**